docs/repositories-configure/using-submodules.md

```diff
@@ -15,12 +15,10 @@ After we enabled submodules for your organization, do the following:
     -   **Public repositories**, make sure that your Git URL uses the HTTPS protocol
     -   **Private repositories**, make sure that your Git URL uses the SSH protocol.
 
-1.  If you already have repositories on Codacy where you want to use submodules, navigate to the page **Settings** and click the button **Generate New User Key** for each of these repositories.
+1.  If you already have repositories on Codacy where you want to use submodules, add a new [SSH key to your git provider account](#user-account-keys) for each of the repositories. On the repository **Settings** page, you can click the link **Add new user key** to go to your Git provider page where you manage your user account SSH keys.
 
-    !!! important
-        **If you're using Bitbucket Cloud**, you must manually add the SSH key on your [user account](https://bitbucket.org/account/settings/ssh-keys/).
-
-    When you add a repository to Codacy, Codacy adds a new SSH key to your repository using an integration with your Git provider. Codacy uses this SSH key every time it needs to clone that repository. When you're using submodules, Codacy needs to clone additional repositories it has no access to. Since there is no way of identifying the submodules of a repository before cloning the repository, Codacy adds the SSH keys to the user account so that it has access to the same repositories as the user.
+    !!! note
+        **If you're using GitLab Cloud**, navigate to the page **Settings** and click the button **Generate New User Key** for each repository. Codacy generates new keys on your user account.
```

Comment:
This detail seems really "dangling" here just for GitLab. Maybe there's a way to integrate this difference directly in the main instruction?

For example:

> [...] and clicking the link **Add new user key** or the button **Generate New User Key**, depending on your Git provider.
>
> For GitHub and Bitbucket this takes you to the Git provider page of the repository, where you can manage your user account SSH keys. For GitLab, Codacy creates the key on your user account automatically.